workspace_api/views.py

```diff
@@ -98,7 +100,19 @@ async def create_workspace(
         )
     )
 
-    create_bucket(workspace_name=workspace_name)
+    # create_bucket(workspace_name=workspace_name)
+    response = requests.get(bucket_endpoint_url, {...})
+    wait_task = asyncio.create_task(wait_for_secret(response.body))
+    if response.status == 200:
+        # we have a bucket created, we create the secret and continue setting up workspace
+        secret = create_secret(response.body)
+
+    elif response.status == 201:
+        # async approach
+        # wait for the secret to be created
+        secret = await wait_task
+
+    # TODO: create the workspace using the secret
```

Comment:
When we have `200` response code, this means we create the secret ourselves, so we don't have to create a task to wait for the secret. So, this should only go into the `201` branch of the `if`.

Also, waiting for the secret should be done in a background task, see line 121 for that.